olive/engine/engine.py

```diff
@@ -121,7 +121,7 @@ def __init__(
             skip_get_available_ep = False
             if (
                 self.target.system_type in (SystemType.AzureML, SystemType.Docker, SystemType.PythonEnvironment)
-                and self.target.olive_managed_env
+                or self.target.olive_managed_env
```

Comment:
the logic is wrong

Reply:
`or` looks right to me. 
We don't want to use the local `ort.get_available_providers` if the system is not local or managed env. https://github.com/microsoft/Olive/blob/main/olive/hardware/accelerator.py#L71 

Otherwise for non-managed, non-local systems, we would be using the available eps from the local environment.

Reply:
Yeah the reason I made this change just like Jambay's comment. We shouldn't get available eps from local environment for non-managed and non-local system.

Reply:
If we'd like not get available EP for docker and azureml, we need explicitly logic to handle this. The reason Lei add this check is because the python environment will fail since the skip is true and the available EP for python env is not correct.

Reply:
Created this PR to get the correct available eps https://github.com/microsoft/Olive/pull/591 

Most tests pass. Just need to make sure the docker tests pass.